packages/teleport/src/lib/tdp/sharedDirectoryManager.ts

```diff
@@ -49,17 +49,30 @@ export class SharedDirectoryManager {
 
     const fileOrDir = await this.walkPath(path);
 
+    let isEmpty = true;
     if (fileOrDir.kind === 'directory') {
+      for await (const _ of fileOrDir.keys()) {
```

Comment:
🤯 What is this dark magic?

Reply:
🧙‍♂️🪄

Added an explanation in afd9ec832ac19464870578bd0c3959f79409e3e7